### PR TITLE
501 review and document unit testing practices

### PR DESCRIPTION
**Pull request by rstens**
Updates of text and organization of the files related to testing and testing documentation.

### CODE REVIEW

File: docs/testing/unit/README.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+## GUI (Vue)
+
+### Linting
+
+ESLinting is use for the UI Components.
+Linting requires:
+
+```
+"eslint": "^5.16.0",
+"eslint-plugin-vue": "^5.0.0",
```
Comment (WalterMoar) [inline docs/testing/unit/README.md]: Note that these will be a rapidly moving target.

File: docs/testing/unit/README.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    "test:unit": "vue-cli-service test:unit",
+    "test:unit:cov": "vue-cli-service test:unit --coverage",
+    "test:e2e": "vue-cli-service test:e2e",
+...
+  }
+```
+
+### Code Coverage Setup
+
+Addign code coverage is simply done by adding `--coverage` to the test command resulting in:
```
Comment (WalterMoar) [inline docs/testing/unit/README.md]: Small typo in `Addign`.

**Review by WalterMoar — APPROVED** [review]
One small typo but otherwise awesome.

File: docs/testing/unit/README.md (hunk 1 above)
Comment (rstens) [reply docs/testing/unit/README.md]: Yes, I'll remove the version numbers and add a comment related to keeping version up-to-date

File: docs/testing/unit/README.md (hunk 2 above)
Comment (rstens) [reply docs/testing/unit/README.md]: Fixed

**Pull request merged**